components/hab/src/command/pkg/exec.rs

```diff
@@ -11,12 +11,26 @@ use crate::hcore::{fs::{find_command,
 use crate::error::{Error,
                    Result};
 
+const PATH_KEY: &str = "PATH";
+
 pub fn start<T>(ident: &PackageIdent, command: T, args: &[OsString]) -> Result<()>
     where T: Into<PathBuf>
 {
     let command = command.into();
     let pkg_install = PackageInstall::load(&ident, Some(&*FS_ROOT_PATH))?;
-    let cmd_env = pkg_install.environment_for_command()?;
+    let mut cmd_env = pkg_install.environment_for_command()?;
+
+    if let Some(path) = cmd_env.get(PATH_KEY) {
+        if let Some(val) = env::var_os(PATH_KEY) {
+            let mut paths: Vec<PathBuf> = env::split_paths(&path).collect();
+            let mut os_paths = env::split_paths(&val).collect();
+            paths.append(&mut os_paths);
+            let joined = env::join_paths(paths)?;
+            let path_str = joined.into_string()
+                                 .expect("Unable to convert OsStr path to string!");
```

Comment:
Could we do something like `.map_err(|s| io::Error::new(io::ErrorKind::InvalidData, s.to_string_lossy()))?;` instead of `expect`?